scripts/tasks.sh

```diff
@@ -718,6 +724,97 @@ function install_host_dependencies_longhorn() {
     longhorn_host_init_common "${DIR}/packages/host/longhorn"
 }
 
+function weave_to_flannel_primary() {
+    shift
+    while [ "$1" != "" ]; do
+        _param="$(echo "$1" | cut -d= -f1)"
+        _value="$(echo "$1" | grep '=' | cut -d= -f2-)"
+        case $_param in
+            cert-key)
+                CERT_KEY="$_value"
+                ;;
+            *)
+                echo >&2 "Error: unknown parameter \"$_param\""
+                exit 1
+                ;;
+        esac
+        shift
+    done
+
+    task_requires_root
+
+    # if CERT_KEY was not provided, we cannot continue
+    if [ -z "$CERT_KEY" ]; then
+        bail "cert-key is required"
```

Comment:
could we try to improve the message and add the steps / where / how the CERT KEY should be provided so that if it fails users knows what they must to do?

Reply:
The cert key should always be provided as part of the command that was given for a user to run